pyproject.toml

```diff
@@ -20,9 +20,11 @@ dependencies = [
     "PyMuPDF>=1.24.3",  # python -c "import pymupdf" won't run before version 1.24.3
     "aiohttp",  # TODO: remove in favor of httpx
     "anyio",
+    "coredis",
```

Comment:
We forgot to add these to `mypy` in `pre-commit`, and when running `mypy` locally I am getting a bunch of type errors in `rate_limiter.py` now. @mskarlin sorry to grave dig with this comment, but do you mind adding these to `mypy` deps, and fixing the errors?

Reply:
https://github.com/Future-House/paper-qa/pull/537